src/context/notesContext.js

```diff
@@ -0,0 +1,199 @@
+        if (status === 200)
+          notesDispatch({ type: "SET_DATA", payload: data.notes });
+      } catch (error) {
+        addToast({
+          type: "Error",
+          msg: "Unable to Fetch Data from API",
+        });
+        console.error(error);
+      }
+    };
+    const postNote = async (notes,content) => {
+      console.log(notes)
+      const note = {...notes,description:parse(content).props.children,_id:uuid()}
+      console.log(note)
+      notesDispatch({ type: "ADD_NOTE", payload: note });
+      // try {
+      //   const { status, data } = await privateInstance({
+      //     method: "post",
+      //     url: "/notes",
+      //     data: {
+      //       note
+      //     },
+      //   });
+      //   if (status === 201) {
+      //     notesDispatch({ type: "ADD_NOTE", payload: note });
+      //     return data.notes;
+      //   }
+      // } catch (error) {
+      //   addToast({
+      //     type: "Error",
+      //     msg: "Unable to Add Note",
+      //   });
+      //   console.error(error);
+      // }
+    };
+    const deleteNote = async (id) => {
+      try {
+        const { status, data } = await privateInstance({
+          method: "delete",
+          url: `/notes/${id}`,
+        });
+        if (status === 200) {
+          notesDispatch({ type: "REMOVE_NOTE", payload: id });
+          return data.notes;
+        }
+      } catch (error) {
+        addToast({
+          type: "Error",
+          msg: "Unable to Delete Note",
+        });
+        console.error(error);
+      }
+    };
+    const updateNote = async (notes, content) => {
+      const note = {
+        ...notes,
+        description: parse(content).props.children
+      };
+      console.log(note)
+      notesDispatch({ type: "UPDATE_NOTE", payload:note});
+      // try {
+      //   const { status, data } = await privateInstance({
+      //     method: "post",
+      //     url: `/notes/${id}`,
+      //   });
+      //   if (status === 200) {
+      //     notesDispatch({ type: "UPDATE_NOTE", payload: id });
+      //     return data.notes;
+      //   }
+      // } catch (error) {
+      //   addToast({
+      //     type: "Error",
+      //     msg: "Unable to Update Note",
+      //   });
+      //   console.error(error);
+      // }
+    };
+
+    const getArchiveNotes = async () => {
+      try {
+        const { status, data } = await privateInstance.get("/archives");
+        if (status === 200)
+          archiveNotesDispatch({
+            type: "SET_ARCHIVE_DATA",
+            payload: data.archives,
+          });
+      } catch (error) {
+        addToast({
+          type: "Error",
+          msg: "Unable to Fetch Data from API",
+        });
+        console.error(error);
+      }
+    };
+    const addNoteToArchive = async (note) => {
+      console.log(note);
+      notesDispatch({ type: "ADD_NOTE", payload: note });
+      archiveNotesDispatch({ type: "REMOVE_NOTE", payload: note._id });
```

Comment:
why these two are getting dispatch here ? should be after success response.

Reply:
Right Now i am getting 500 in API.thatwhy bypassing it right now